src/pylint_datetime/__init__.py

```diff
@@ -1,4 +1,185 @@
-# License: MIT
-# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+"""Pylint checker ensure correct usage of datetime module, especially around aware/naive objects."""
 
-"""checker to ensure correct datetime usage with aware objects in datetime module."""
+from typing import Any
+
+import astroid  # type: ignore
+import pylint  # type: ignore
+from pylint.checkers import BaseChecker  # type: ignore
+
+# function calls to fromisoformat, strptime and fromisoformat can't be checked - they parse strings
+# if the specified timezone is None, still allowed although they produce naive objects
+
+
+class DatetimeChecker(BaseChecker):  # type: ignore
+    """class for custom pylint checker."""
+
+    name = "pylint-datetime"
+    priority = -1
+    msgs = {
+        "W9999": (
+            "timedelta() called without keyword arguments",
+            "timedelta-no-keyword-args",
+            "timedelta() called without keyword arguments.",
+        ),
+        "W9998": (
+            'Function call to "%s"/"replace" should be called with a timezone argument',
```

Comment:
Why the `/"replace"` here?

Reply:
it isn't supposed to be be, I forget to remove if and pass replace as a parameter to then insert it into %s